jsf/WFLY-13350_galleon_layer_for_jsf_support.adoc

```diff
@@ -0,0 +1,104 @@
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] which will be responsible for providing the WildFly JSF subsystem.
+
+The following is a configuration example of the WildFly bootable JAR Maven plugin to provision a server with the `jsf` layer:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>jsf</layer>
+        </layers>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13350[WFLY-13350 Add a Galleon layer for JSF]
+* https://issues.redhat.com/browse/EAP7-1529[EAP7-1529 Add a Galleon layer for JSF]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-13354[WFLY-13354]
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* The Galleon layer must provide a servlet container and configure the WildFly JSF subsystem.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+* The JSF layer does not expose a management interface. That means, it doesn't need to depend on any security subsystems.
+
+== Test Plan
+
+The test coverage of the Galleon layer added by this proposal is divided in two main groups:
+
+. Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:
+
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
+
+. Execution of WildFly tests related to the feature provisioned by this layer. Reuse the existing tests available on the WildFly test suite, which are directly testing this layer functionalities, and execute them on a server installation provisioned with this layer.
+
+== Community Documentation
+
+Community documentation plan is adding the layer to https://docs.wildfly.org/20/Admin_Guide.html#wildfly-galleon-layers[WildFly Galleon layers] in the section it belongs to.
```

Comment:
@yersan given JSF can be used:
* with or without CDI 
* with or without EJB

does it makes sense to add some quick information on what combination of base layers provides the user with CDI/EJB? I mean e.g. with `datasources-web-server` the user don't have neither CDI or EJB, rigth?

Reply:
Hi @tommaso-borgato, JSF includes CDI but it does not includes EJB3. In general I think the documentation about the layers could be improved to make it more easily understandable, but I would say it is something outside of this RFE.

{quote}
`datasources-web-server` the user doesn't have CDI/EJB, right?
{quote}

Right, neither CDI or EJB are supplied by this layer.